### PR TITLE
[Fix #2683] Do not run sync state loop on upstream RPC networks

### LINKED ISSUE #2683: Do not run sync state loop on upstream RPC networks
We don't need sync state on upstream RPC networks because they do not need to sync

### PR DESCRIPTION
**Pull request by foopang**
fixes #2683 

status: ready

### CODE REVIEW

File: src/status_im/protocol/handlers.cljs [hunk 1]
```diff
@@ -15,7 +15,8 @@
             [taoensso.timbre :as log :refer-macros [debug]]
             [status-im.native-module.core :as status]
             [clojure.string :as string]
-            [status-im.utils.web3-provider :as web3-provider]))
+            [status-im.utils.web3-provider :as web3-provider]
+            [status-im.ui.screens.network-settings.events :refer [network-with-upstream-rpc?]]))
```
Comment (flexsurfer) [inline src/status_im/protocol/handlers.cljs]: better to move this function to utils, and use `:as utils` instead `:refer`

File: src/status_im/protocol/handlers.cljs [hunk 2]
```diff
@@ -302,8 +303,8 @@
 
 (handlers/register-handler-fx
   :initialize-sync-listener
-  (fn [{{:keys [sync-listening-started] :as db} :db} _]
-    (when-not sync-listening-started
+  (fn [{{:keys [sync-listening-started network] networks :networks/networks :as db} :db} _]
+    (when-not (or sync-listening-started (network-with-upstream-rpc? networks network))
```
Comment (flexsurfer) [inline src/status_im/protocol/handlers.cljs]: `when-not` with `or` is difficult to read could you please change it on `when`

**Review by flexsurfer — CHANGES_REQUESTED** [review]
thanks for contribution @foopang , just little comments

**Comment by foopang** [comment]
@flexsurfer  Updated. Thanks for your review. I don't know whether if there is an appropriate namespace already in existence that I could use for the util function `network-with-upstream-rpc?`. So I created a new namespace called `status-im.protocol.utils` for it, hope this makes sense to you.

**Comment by flexsurfer** [comment]
thanks @foopang , sorry haven't said , please move this function to `status-im.utils.ethereum.core`, sorry for confusion

**Comment by foopang** [comment]
@flexsurfer updated.

**Review by flexsurfer — APPROVED** [review]

**Pull request merged**